### PR TITLE
Trim leading whitespace from code snippets

### PR DESCRIPTION
**Pull request by JssDWt**
Adds a mdbook preprocessor that trims leading whitespace from the code snippets. It removes the indentation basically if every line in a code snippet has that indentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -13,6 +13,7 @@ To locally serve the docs run:
 
 ```bash
 cargo install mdbook
+cargo install ./snippets-processor
```
Comment (ok300) [inline README.md:16]:
```suggestion
cargo install --path ./snippets-processor
```

**Comment by ok300** [comment]
Got it to work locally, very cool!

Reading the docs flows much better now.

**Review by ok300 — APPROVED** [review]
Found one nit about `--path` arg, see above.

Since this needs the new crate to be installed, I also wondered if it would be picked up by the CI when it builds the docs. It probably needs to be installed there too. I didn't find the place where this is done, will ask @yaslama on Monday.

Otherwise looks good! Let's merge, it fixes indentation for all snippet languages.

**Comment by JssDWt** [comment]
> Since this needs the new crate to be installed, I also wondered if it would be picked up by the CI when it builds the docs. It probably needs to be installed there too. I didn't find the place where this is done, will ask @yaslama on Monday.

We do have to change that, it's here currently: https://github.com/breez/breez-sdk/blob/main/.github/workflows/rustdoc-pages.yml

File: README.md (hunk 1 above)
Comment (JssDWt) [reply README.md]: Done

**Comment by ok300** [comment]
I don't know if thats the workflow building the mdbook, looks like its the one doing [the rustdocs](https://breez.github.io/breez-sdk/breez_sdk_core/)?

**Comment by JssDWt** [comment]
> I don't know if thats the workflow building the mdbook, looks like its the one doing [the rustdocs](https://breez.github.io/breez-sdk/breez_sdk_core/)?

Ah yeah you're right.

**Comment by JssDWt** [comment]
Added a compilation step in the CI, since we're using a custom preprocessor now.

**Pull request merged**